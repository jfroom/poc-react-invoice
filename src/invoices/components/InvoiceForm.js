import React, {Component} from 'react';
import PropTypes from 'prop-types'
import {
  Button, Table, FormGroup, ControlLabel, FormControl, Glyphicon
} from 'react-bootstrap';
import {LinkContainer} from 'react-router-bootstrap';
import NumberFormat from 'react-number-format'

class InvoiceForm extends Component {
  render() {
    const invoiceEmpty = {title: '', status: '', total: 0, items: []}
    const invoice = this.props.invoice ? this.props.invoice : invoiceEmpty

    const renderItems = () => {
      return (
        invoice.items.map((item, idx) =>
          <tr key={idx}>
            <td>{item.text}</td>
            <td><NumberFormat value={item.price} displayType={'text'} thousandSeparator={true} decimalPrecision={true} prefix={'$'} /></td>
            <td>
              <Button className='btn-xs btn-danger'><Glyphicon glyph='trash'/></Button>
            </td>
          </tr>
        )
      )
    }

    return (
      <div className="InvoiceForm">
        <h1>{this.props.action} Invoice</h1>
        <hr/>
        <form>
          <div className='row'>
            <div className='col-sm-10'>
              <FormGroup controlId='title' validationState={null}>
                <ControlLabel>Title</ControlLabel>
                <FormControl type='text' value={invoice.title} placeholder='Enter title' onChange={() => {}}/>
                <FormControl.Feedback/>
              </FormGroup>
            </div>
            <div className='col-sm-2'>
              <FormGroup controlId='status' validationState={null}>
                <ControlLabel>Status</ControlLabel>
                <FormControl componentClass="select" placeholder="Status" value={invoice.status} onChange={() => {}}>
                  <option value="due">Due</option>
                  <option value="overdue">Overdue</option>
                  <option value="paid">Paid</option>
                </FormControl>
              </FormGroup>
            </div>
          </div>


          <FormGroup controlId='line-items' validationState={null}>
            <ControlLabel>Line Items</ControlLabel>
            <div className='well'>
              <Table condensed className='line-items'>
                <thead>
                  <tr>
                    <th>Description</th>
                    <th>Price</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  {renderItems()}
                  <tr>
                    <td>
                      <FormGroup controlId={'text-0'} validationState={null}>
                        <FormControl type='text' value='' placeholder="Enter line item"/>
                        <FormControl.Feedback/>
                      </FormGroup>
                    </td>
                    <td>
                      <FormGroup controlId={'cost-0'} validationState={null}>
                        <FormControl type='text' value='' placeholder="Enter price"/>
                        <FormControl.Feedback/>
                      </FormGroup>
                    </td>
                    <td>
                      <Button className='btn-xs btn-success'><Glyphicon glyph='plus'/> Add</Button>
                    </td>
                  </tr>
                </tbody>
              </Table>
              </div>
            </FormGroup>

          <FormGroup controlId='total' validationState={null}>
            <ControlLabel>Total Due</ControlLabel>
            <FormControl.Static>
              <NumberFormat value={invoice.total} displayType={'text'} thousandSeparator={true} decimalPrecision={true} prefix={'$'} />
            </FormControl.Static>
            <FormControl.Feedback/>
          </FormGroup>

          <FormGroup controlId='notes' validationState={null}>
            <ControlLabel>Notes</ControlLabel>
            <FormControl componentClass='textArea' value={invoice.notes} placeholder='Enter notes' onChange={() => {}}/>
            <FormControl.Feedback/>
          </FormGroup>

          <div className='text-right'>
            <LinkContainer to="/" className='btn-default btn-lg' exact>
              <Button>Cancel</Button>
            </LinkContainer>
            <Button type="submit" className='btn-success btn-lg'>Save</Button>
          </div>
        </form>
        <p/>
      </div>
    )
  }
}

InvoiceForm.propTypes = {
  invoice: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    total: PropTypes.number.isRequired,
    items: PropTypes.arrayOf(PropTypes.shape({
      text: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired
    }).isRequired).isRequired,
    notes: PropTypes.text
  }).isRequired
}
export default InvoiceForm
