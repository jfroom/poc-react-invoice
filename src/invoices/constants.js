// @flow
export const StatusTypes = {
  NONE: '',
  PAID: 'Paid',
  UNPAID: 'Unpaid',
}
