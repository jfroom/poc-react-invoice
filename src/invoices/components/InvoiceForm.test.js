import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import InvoiceForm from './InvoiceForm'
import { StatusTypes } from '../constants'
import { BrowserRouter as Router } from 'react-router-dom'

let props = {}
let handleAddInvoice
let handleEditInvoice

const setField = (wrap, id, val) => {
  let input = wrap.find(`#${id}`).first()
  input.node.value = val
  input.simulate('change', input)
}

const testInvoice = {
  title: 'Test title',
  date: '1/2/2017',
  status: StatusTypes.PAID,
  notes: 'Test notes',
  items: [
    { text: 'test item 1', price: 7.77 },
    { text: 'test item 1', price: 1.21 }
  ]
}
beforeEach( () => {
  handleAddInvoice = jest.fn()
  handleEditInvoice = jest.fn()
  props = { handleAddInvoice, handleEditInvoice }
})

it('new invoice', () => {
  // Mount & verify snapshot
  const wrap = mount(<Router><InvoiceForm {...props} /></Router>)
  expect(toJson(wrap)).toMatchSnapshot()
  expect(wrap.find('h1')).toHaveText('New Invoice')

  // Populate fields
  let invoice = { ...testInvoice, id: -1 }
  const fields = ['title', 'date', 'status', 'notes']
  fields.forEach( field => {
    setField(wrap, field, invoice[field])
  })

  // Populate line-item fields
  invoice.items.forEach((item) => {
    setField(wrap, 'lineItemText', item.text)
    setField(wrap, 'lineItemPrice', item.price)
    wrap.find('#add-item-btn').first().simulate('click')
  })

  // Submit & verify
  wrap.find('#form').first().simulate('submit')
  invoice.total = 8.98
  expect(handleAddInvoice.mock.calls).toEqual([[invoice]])
})

it('edit invoice', () => {
  let invoice = { ...testInvoice, id: 1 }
  props.invoice = invoice

  // Mount & verify snapshot
  const wrap = mount(<Router><InvoiceForm {...props} /></Router>)
  expect(toJson(wrap)).toMatchSnapshot()
  expect(wrap.find('h1').first()).toHaveText('Edit Invoice')

  // Change title
  invoice.title = "edited title"
  setField(wrap, 'title', invoice.title)

  // Remove first line item
  wrap.find('.btn-delete-item').first().simulate('click')
  invoice.total = 1.21

  // Submit & verify
  wrap.find('#form').first().simulate('submit')
  invoice.items.splice(0, 1)
  expect(handleEditInvoice.mock.calls).toEqual([[invoice]])
})
