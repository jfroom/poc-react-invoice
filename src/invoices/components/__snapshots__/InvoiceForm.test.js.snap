// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit invoice 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InvoiceForm
      handleAddInvoice={[Function]}
      handleEditInvoice={[Function]}
      invoice={
        Object {
          "date": "1/2/2017",
          "id": 1,
          "items": Array [
            Object {
              "price": 7.77,
              "text": "test item 1",
            },
            Object {
              "price": 1.21,
              "text": "test item 1",
            },
          ],
          "notes": "Test notes",
          "status": "Paid",
          "title": "Test title",
        }
      }
    >
      <styled.div>
        <div
          className="sc-bdVaJa hdXCUo"
        >
          <h1>
            Edit
             Invoice
          </h1>
          <hr />
          <form
            id="form"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col-sm-8"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="title"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="title"
                      >
                        Title
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Enter title"
                      type="text"
                      value="Test title"
                    >
                      <input
                        className="form-control"
                        id="title"
                        onChange={[Function]}
                        placeholder="Enter title"
                        type="text"
                        value="Test title"
                      />
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
              <div
                className="col-sm-2"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="date"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="date"
                      >
                        Due Date
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Ex. 1/1/2017"
                      type="text"
                      value="1/2/2017"
                    >
                      <input
                        className="form-control"
                        id="date"
                        onChange={[Function]}
                        placeholder="Ex. 1/1/2017"
                        type="text"
                        value="1/2/2017"
                      />
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
              <div
                className="col-sm-2"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="status"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="status"
                      >
                        Status
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="select"
                      onChange={[Function]}
                      value="Paid"
                    >
                      <select
                        className="form-control"
                        id="status"
                        onChange={[Function]}
                        value="Paid"
                      >
                        <option
                          value="Unpaid"
                        >
                          Unpaid
                        </option>
                        <option
                          value="Paid"
                        >
                          Paid
                        </option>
                      </select>
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
            </div>
            <FormGroup
              bsClass="form-group"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                  >
                    Line Items
                  </label>
                </ControlLabel>
                <div
                  className="well"
                >
                  <Table
                    bordered={false}
                    bsClass="table"
                    condensed={false}
                    hover={false}
                    responsive={false}
                    striped={false}
                  >
                    <table
                      className="table"
                    >
                      <thead>
                        <tr>
                          <th>
                            Description
                          </th>
                          <th>
                            Price
                          </th>
                          <th>
                            Action
                          </th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr>
                          <td>
                            test item 1
                          </td>
                          <td>
                            $7.77
                          </td>
                          <td>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              className="btn-xs btn-danger btn-delete-item"
                              disabled={false}
                              onClick={[Function]}
                            >
                              <button
                                className="btn-xs btn-danger btn-delete-item btn btn-default"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="trash"
                                >
                                  <span
                                    className="glyphicon glyphicon-trash"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                          </td>
                        </tr>
                        <tr>
                          <td>
                            test item 1
                          </td>
                          <td>
                            $1.21
                          </td>
                          <td>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              className="btn-xs btn-danger btn-delete-item"
                              disabled={false}
                              onClick={[Function]}
                            >
                              <button
                                className="btn-xs btn-danger btn-delete-item btn btn-default"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="trash"
                                >
                                  <span
                                    className="glyphicon glyphicon-trash"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                          </td>
                        </tr>
                        <tr>
                          <td>
                            <FormGroup
                              bsClass="form-group"
                              controlId="lineItemText"
                            >
                              <div
                                className="form-group"
                              >
                                <FormControl
                                  bsClass="form-control"
                                  componentClass="input"
                                  id="lineItemText"
                                  onChange={[Function]}
                                  placeholder="Enter line item"
                                  type="text"
                                  value=""
                                >
                                  <input
                                    className="form-control"
                                    id="lineItemText"
                                    onChange={[Function]}
                                    placeholder="Enter line item"
                                    type="text"
                                    value=""
                                  />
                                </FormControl>
                              </div>
                            </FormGroup>
                          </td>
                          <td>
                            <FormGroup
                              bsClass="form-group"
                              controlId="lineItemPrice"
                            >
                              <div
                                className="form-group"
                              >
                                <FormControl
                                  bsClass="form-control"
                                  componentClass="input"
                                  min="0"
                                  onChange={[Function]}
                                  placeholder="Enter price"
                                  step="any"
                                  type="number"
                                  value=""
                                >
                                  <input
                                    className="form-control"
                                    id="lineItemPrice"
                                    min="0"
                                    onChange={[Function]}
                                    placeholder="Enter price"
                                    step="any"
                                    type="number"
                                    value=""
                                  />
                                </FormControl>
                              </div>
                            </FormGroup>
                          </td>
                          <td>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              className="btn-xs btn-success"
                              disabled={false}
                              id="add-item-btn"
                              onClick={[Function]}
                            >
                              <button
                                className="btn-xs btn-success btn btn-default"
                                disabled={false}
                                id="add-item-btn"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="plus"
                                >
                                  <span
                                    className="glyphicon glyphicon-plus"
                                  />
                                </Glyphicon>
                                 Add
                              </button>
                            </Button>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </Table>
                </div>
              </div>
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              controlId="total"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="total"
                  >
                    Total Due
                  </label>
                </ControlLabel>
                <FormControlStatic
                  bsClass="form-control-static"
                  componentClass="p"
                >
                  <p
                    className="form-control-static"
                  >
                    $0.00
                  </p>
                </FormControlStatic>
              </div>
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              controlId="notes"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="notes"
                  >
                    Notes
                  </label>
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="textArea"
                  onChange={[Function]}
                  placeholder="Enter notes"
                  value="Test notes"
                >
                  <textArea
                    className="form-control"
                    id="notes"
                    onChange={[Function]}
                    placeholder="Enter notes"
                    value="Test notes"
                  />
                </FormControl>
              </div>
            </FormGroup>
            <div
              className="text-right"
            >
              <LinkContainer
                activeClassName="active"
                className="btn-default btn-lg"
                exact={true}
                replace={false}
                strict={false}
                to="/"
              >
                <Route
                  exact={true}
                  path="/"
                  strict={false}
                >
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="btn-default btn-lg active"
                    disabled={false}
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <SafeAnchor
                      className="btn-default btn-lg active btn btn-default"
                      componentClass="a"
                      disabled={false}
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <a
                        className="btn-default btn-lg active btn btn-default"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Cancel
                      </a>
                    </SafeAnchor>
                  </Button>
                </Route>
              </LinkContainer>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="btn-success btn-lg"
                disabled={false}
                id="submit"
                type="submit"
              >
                <button
                  className="btn-success btn-lg btn btn-default"
                  disabled={false}
                  id="submit"
                  type="submit"
                >
                  Save
                </button>
              </Button>
            </div>
          </form>
          <p />
        </div>
      </styled.div>
    </InvoiceForm>
  </Router>
</BrowserRouter>
`;

exports[`new invoice 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InvoiceForm
      handleAddInvoice={[Function]}
      handleEditInvoice={[Function]}
      invoice={
        Object {
          "date": "",
          "id": -1,
          "items": Array [],
          "notes": "",
          "status": "Unpaid",
          "title": "",
          "total": 0,
        }
      }
    >
      <styled.div>
        <div
          className="sc-bdVaJa hdXCUo"
        >
          <h1>
            New
             Invoice
          </h1>
          <hr />
          <form
            id="form"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col-sm-8"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="title"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="title"
                      >
                        Title
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Enter title"
                      type="text"
                      value=""
                    >
                      <input
                        className="form-control"
                        id="title"
                        onChange={[Function]}
                        placeholder="Enter title"
                        type="text"
                        value=""
                      />
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
              <div
                className="col-sm-2"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="date"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="date"
                      >
                        Due Date
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Ex. 1/1/2017"
                      type="text"
                      value=""
                    >
                      <input
                        className="form-control"
                        id="date"
                        onChange={[Function]}
                        placeholder="Ex. 1/1/2017"
                        type="text"
                        value=""
                      />
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
              <div
                className="col-sm-2"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="status"
                >
                  <div
                    className="form-group"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      <label
                        className="control-label"
                        htmlFor="status"
                      >
                        Status
                      </label>
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="select"
                      onChange={[Function]}
                      value="Unpaid"
                    >
                      <select
                        className="form-control"
                        id="status"
                        onChange={[Function]}
                        value="Unpaid"
                      >
                        <option
                          value="Unpaid"
                        >
                          Unpaid
                        </option>
                        <option
                          value="Paid"
                        >
                          Paid
                        </option>
                      </select>
                    </FormControl>
                  </div>
                </FormGroup>
              </div>
            </div>
            <FormGroup
              bsClass="form-group"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                  >
                    Line Items
                  </label>
                </ControlLabel>
                <div
                  className="well"
                >
                  <Table
                    bordered={false}
                    bsClass="table"
                    condensed={false}
                    hover={false}
                    responsive={false}
                    striped={false}
                  >
                    <table
                      className="table"
                    >
                      <thead>
                        <tr>
                          <th>
                            Description
                          </th>
                          <th>
                            Price
                          </th>
                          <th>
                            Action
                          </th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr>
                          <td>
                            <FormGroup
                              bsClass="form-group"
                              controlId="lineItemText"
                            >
                              <div
                                className="form-group"
                              >
                                <FormControl
                                  bsClass="form-control"
                                  componentClass="input"
                                  id="lineItemText"
                                  onChange={[Function]}
                                  placeholder="Enter line item"
                                  type="text"
                                  value=""
                                >
                                  <input
                                    className="form-control"
                                    id="lineItemText"
                                    onChange={[Function]}
                                    placeholder="Enter line item"
                                    type="text"
                                    value=""
                                  />
                                </FormControl>
                              </div>
                            </FormGroup>
                          </td>
                          <td>
                            <FormGroup
                              bsClass="form-group"
                              controlId="lineItemPrice"
                            >
                              <div
                                className="form-group"
                              >
                                <FormControl
                                  bsClass="form-control"
                                  componentClass="input"
                                  min="0"
                                  onChange={[Function]}
                                  placeholder="Enter price"
                                  step="any"
                                  type="number"
                                  value=""
                                >
                                  <input
                                    className="form-control"
                                    id="lineItemPrice"
                                    min="0"
                                    onChange={[Function]}
                                    placeholder="Enter price"
                                    step="any"
                                    type="number"
                                    value=""
                                  />
                                </FormControl>
                              </div>
                            </FormGroup>
                          </td>
                          <td>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              className="btn-xs btn-success"
                              disabled={false}
                              id="add-item-btn"
                              onClick={[Function]}
                            >
                              <button
                                className="btn-xs btn-success btn btn-default"
                                disabled={false}
                                id="add-item-btn"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="plus"
                                >
                                  <span
                                    className="glyphicon glyphicon-plus"
                                  />
                                </Glyphicon>
                                 Add
                              </button>
                            </Button>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </Table>
                </div>
              </div>
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              controlId="total"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="total"
                  >
                    Total Due
                  </label>
                </ControlLabel>
                <FormControlStatic
                  bsClass="form-control-static"
                  componentClass="p"
                >
                  <p
                    className="form-control-static"
                  >
                    $0.00
                  </p>
                </FormControlStatic>
              </div>
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              controlId="notes"
            >
              <div
                className="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="notes"
                  >
                    Notes
                  </label>
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="textArea"
                  onChange={[Function]}
                  placeholder="Enter notes"
                  value=""
                >
                  <textArea
                    className="form-control"
                    id="notes"
                    onChange={[Function]}
                    placeholder="Enter notes"
                    value=""
                  />
                </FormControl>
              </div>
            </FormGroup>
            <div
              className="text-right"
            >
              <LinkContainer
                activeClassName="active"
                className="btn-default btn-lg"
                exact={true}
                replace={false}
                strict={false}
                to="/"
              >
                <Route
                  exact={true}
                  path="/"
                  strict={false}
                >
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="btn-default btn-lg active"
                    disabled={false}
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <SafeAnchor
                      className="btn-default btn-lg active btn btn-default"
                      componentClass="a"
                      disabled={false}
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <a
                        className="btn-default btn-lg active btn btn-default"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Cancel
                      </a>
                    </SafeAnchor>
                  </Button>
                </Route>
              </LinkContainer>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="btn-success btn-lg"
                disabled={false}
                id="submit"
                type="submit"
              >
                <button
                  className="btn-success btn-lg btn btn-default"
                  disabled={false}
                  id="submit"
                  type="submit"
                >
                  Save
                </button>
              </Button>
            </div>
          </form>
          <p />
        </div>
      </styled.div>
    </InvoiceForm>
  </Router>
</BrowserRouter>
`;
